From afbd2942b780f176e60f208638f2789d42bd6273 Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Tue, 28 May 2019 16:17:09 -0700
Subject: [PATCH 0916/1239] mmc: octeontx: Skip calibration check on broken hw

The calibration code improperly checks the calibration
status when calibration is not performed on certain
devices that have broken calibration.

Change-Id: I5034fe7eeacb86498f69cdd15a59cb375a29e03b
Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/10131
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
---
 drivers/mmc/octeontx_hsmmc.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/mmc/octeontx_hsmmc.c b/drivers/mmc/octeontx_hsmmc.c
index d0f5519247..9382746b24 100644
--- a/drivers/mmc/octeontx_hsmmc.c
+++ b/drivers/mmc/octeontx_hsmmc.c
@@ -2383,11 +2383,12 @@ static int octeontx_mmc_host_calibrate_delay(struct octeontx_mmc_host *host)
 			udelay(5);
 			emm_tap.u = readq(host->base_addr + CAVM_MIO_EMM_TAP());
 		} while (!emm_tap.s.delay && get_timer(start) < 10);
-	}
-	if (!emm_tap.s.delay) {
-		pr_err("%s: Error: delay calibration failed, timed out.\n",
-		       __func__);
-		return -1;
+
+		if (!emm_tap.s.delay) {
+			pr_err("%s: Error: delay calibration failed, timed out.\n",
+			       __func__);
+			return -1;
+		}
 	}
 
 	host->timing_taps = 10 * 1000 * emm_tap.s.delay / 512;
-- 
2.29.0

